### ISSUE TITLE
Uncaught exception 'Exception' with message 'Reading while at end of stream'

### ISSUE THREAD

**Issue by laapsaap** [mauritsl/php-cassandra#2]
[root@48cores lib]# php test.php 
PHP Fatal error:  Uncaught exception 'Exception' with message 'Reading while at end of stream' in /tmp/php-cassandra/lib/DataStream.php:33
Stack trace:
#0 /tmp/php-cassandra/lib/DataStream.php(75): Cassandra\DataStream->read(30569)
#1 /tmp/php-cassandra/lib/Transport.php(93): Cassandra\DataStream->readString()
#2 /tmp/php-cassandra/lib/Cassandra.php(123): Cassandra\Transport->fetchFrame()
#3 /tmp/php-cassandra/lib/bench.php(10): Cassandra\Connection->query('SELECT \* FROM u...')
#4 {main}

  thrown in /tmp/php-cassandra/lib/DataStream.php on line 33

---

<?php

```
require('./Cassandra.php');

use Cassandra\Connection as Cassandra;

$connection = new Cassandra('10.17.152.18', 'benchbin');

$rows = $connection->query("SELECT * FROM users");
$rows = $connection->query("SELECT * FROM users");
```

**Comment by mauritsl** [comment]
Hm that was fast.
Could you post the table structure of users?

**Comment by laapsaap** [comment]
Didnt do anything special with the table;

CREATE TABLE benchbin.users (
  user_name text PRIMARY KEY,
  birth_year bigint,
  gender text,
  password text,
  session_token text,
  state text
) WITH
  bloom_filter_fp_chance=0.010000 AND
  caching='KEYS_ONLY' AND
  comment='' AND
  dclocal_read_repair_chance=0.000000 AND
  gc_grace_seconds=864000 AND
  read_repair_chance=0.100000 AND
  replicate_on_write='true' AND
  populate_io_cache_on_flush='false' AND
  compaction={'class': 'SizeTieredCompactionStrategy'} AND
  compression={'sstable_compression': 'LZ4Compressor'};

**Comment by mauritsl** [comment]
I don't get an error myself when running that query with that table.
At least I can say that it is trying to read an error message. It just got an error frame back from Cassandra. The body of that frame should consist of an int and a string according to the specifications. Looks like not every error code has an error string, but that's not very clear in the docs.

**Comment by laapsaap** [comment]
What version do you use? I use cassandra20-2.0.3 in this testcluster.

**Comment by mauritsl** [comment]
That version is released just one week ago. I will come back when I have tested that version.

**Comment by mauritsl** [comment]
I have tested the same code on 2.0.3, but still without errors.
My best guess is that not all errors from Cassandra have an error string. I changed the error handling here. We will only read the error string for the 'invalid' and 'syntax' errors. Could you get a normal error now?

https://github.com/mauritsl/php-cassandra/commit/c3145dcf2d8a9b546eca9f964c1359731bf7f7c9

**Comment by mauritsl** [comment]
@laapsaap 
Are you still on this of did you proceed with another driver?

**Comment by laapsaap** [comment]
We are still in the progress testing between php-cassandra, phpcassa and the pdo driver. We did some benchmark a week a go and saw big difference between php-cassandra and phpcassa in favor for phpcassa. Not sure why there's such a big difference. From what I recall 10fold difference.

I am not sure what is wrong here, but I am getting errors with the latest commit. These scripts worked before.

PHP Fatal error:  Uncaught exception 'Exception' with message 'Unknown error' in /root/cassandra-bench/php-cassandra/lib/Transport.php:131
Stack trace:
#0 /root/cassandra-bench/php-cassandra/lib/Transport.php(92): Cassandra\Transport->readError(Object(Cassandra\DataStream))
#1 /root/cassandra-bench/php-cassandra/lib/Cassandra.php(123): Cassandra\Transport->fetchFrame()
#2 /root/cassandra-bench/php-cassandra/lib/bench.php(36): Cassandra\Connection->query('SELECT \* FROM u...')

php version?

# php -v

PHP 5.4.22 (cli) (built: Nov 15 2013 10:13:25) 
Copyright (c) 1997-2013 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2013 Zend Technologies

**Comment by zhil** [comment]
@laapsaap I have tested phpcassa as well - php-cassandra way more simple, flexible and way more fast on Windows. It worked very slowly on our test Ubuntu server, but we have patched it in PR #3

**Comment by mauritsl** [comment]
Merge request https://github.com/mauritsl/php-cassandra/pull/3 is committed. Closing this issue.
